src/screens/loading.rs

```diff
@@ -28,15 +28,34 @@ fn show_loading_screen(mut commands: Commands) {
 
 fn all_assets_loaded(
     asset_server: Res<AssetServer>,
-    image_handles: Res<HandleMap<ImageKey>>,
-    sfx_handles: Res<HandleMap<SfxKey>>,
-    soundtrack_handles: Res<HandleMap<SoundtrackKey>>,
+    image_handles: Res<ImageHandles>,
+    sound_effects_handles: Res<SoundEffectHandles>,
+    soundtrack_handles: Res<SoundtrackHandles>,
 ) -> bool {
     image_handles.all_loaded(&asset_server)
-        && sfx_handles.all_loaded(&asset_server)
+        && sound_effects_handles.all_loaded(&asset_server)
         && soundtrack_handles.all_loaded(&asset_server)
 }
 
 fn continue_to_title(mut next_screen: ResMut<NextState<Screen>>) {
     next_screen.set(Screen::Title);
 }
+
+trait AssetMap {
```

Comment:
Would add a brief doc comment here even though it's not `pub`.
```suggestion
/// An extension trait to check if all the assets in an asset collection are loaded.
trait AssetMap {
```